tket/src/PauliGraph/PauliGraph.cpp

```diff
@@ -113,13 +113,12 @@ void PauliGraph::apply_gate_at_end(
     case OpType::CY:
     case OpType::CZ:
     case OpType::SWAP:
-    case OpType::noop:
-    case OpType::Phase: {
```

Comment:
Can we still handle the global-phase op?

Reply:
It isn't listed in the accepted gates in the preconditions for the PauliGraph passes, and they currently do not preserve global phase so I don't see much value in trying to handle them at this point. I could allow them to be treated as a noop, but the fact that a programmer is using a Phase gate explicitly tells me that they are interested in determining semantics with global phase, so silently removing it without error or warning could be frustrating for them.

Reply:
Ok, I have added Phase gate handling back in. Did not know it was implicitly allowed by GateSetPredicate, but that makes enough sense for it to be handled